go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -0,0 +1,321 @@
+			klog.Errorf("%s", err)
+		}
+	}
+
+	egressFirewall.Status.EgressFirewallObjectStatus = "EgressFirewall Rules applied"
+	oc.updateEgressFirewallObjectWithRetry(egressFirewall)
+
+}
+
+func (oc *Controller) updateEgressFirewall(oldEgressFirewall, newEgressFirewall *egressfirewallapi.EgressFirewall) {
+	if !reflect.DeepEqual(oldEgressFirewall.Spec, newEgressFirewall.Spec) {
+		oc.deleteEgressFirewall(oldEgressFirewall)
+		oc.addEgressFirewall(newEgressFirewall)
+	}
+}
+
+func (oc *Controller) deleteEgressFirewall(egressFirewall *egressfirewallapi.EgressFirewall) {
+	klog.Infof("Deleting egress Firewall %s in namespace %s", egressFirewall.Name, egressFirewall.Namespace)
+
+	nsInfo, err := oc.waitForNamespaceLocked(egressFirewall.Namespace)
+	if err != nil {
+		klog.Errorf("Failed to wait for namespace %s event (%v)",
+			egressFirewall.Namespace, err)
+	}
+	defer nsInfo.Unlock()
+
+	existingNodes, err := oc.kube.GetNodes()
+	if err != nil {
+		klog.Errorf("Error deleting egressFirewall for namespace %s, cannot get nodes to delete ACLS %v",
+			egressFirewall.Namespace, err)
+		// return without clearing EgressFirewall because none of the rules can be deleted
+		return
+	}
+
+	stdout, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+		"--columns=_uuid,match", "find", "ACL",
+		fmt.Sprintf("external-ids:egressFirewall=%s", egressFirewall.Namespace))
+	if err != nil {
+		klog.Errorf(" error executing find ACL command, stderr: %q, %+v", stderr, err)
+		// return without clearing EgressFirewall because none of the rules can be deleted
+		return
+	}
+	splitStdout := strings.Split(stdout, "\n")
+	// example output from splitStdout
+	// splitStdout = {
+	// "158b96bc-b60b-4fff-b838-f8a5c75af557",
+	// "ip4.src==10.244.1.2",
+	// "",
+	// "0870c6b4-56e4-415f-967d-dde300266931",
+	// "ip4.src==10.244.2.2",
+	// "",
+	// "f10ce608-6286-4f93-9a1f-8b1d92494e17",
+	// "ip4.dst == 0.0.0.0/0 && ip4.src == $a5154718082306775057",
+	// "",
+	// "a785e260-086e-4ac4-b6bf-e2d3cceddce7",
+	// "ip4.src==10.244.0.2",
+	// "",
+	// "53cab5de-5e5b-405c-9915-b1efec69d433",
+	// "ip4.dst == 1.2.3.0/24 && ip4.src == $a5154718082306775057",
+	// }
+	// each pair of values is a single represents a single ACL entry
+	// 158b96bc-b60b-4fff-b838-f8a5c75af557 is the uuid and
+	// ip4.src==10.244.1.2 is the match statment
+	//
+	// in order to process them correctly loop by twos and if the current line is
+	// an empty string skip it, the current index is the UUID of the acl and
+	// index++ will tell us if the rule uses  ipv4 or ipv6 addresses
+	for index := 0; index < len(splitStdout); index = index + 2 {
+		if splitStdout[index] == "" {
+			// between every entry there is a newline, skip the newline
+			index++
+		}
+		uuid := splitStdout[index]
+		var addressSetHash string
+		if strings.Contains(splitStdout[index+1], "ip4") {
+			addressSetHash = nsInfo.addressSet.GetIPv4HashName()
+		} else {
+			addressSetHash = nsInfo.addressSet.GetIPv6HashName()
```

Comment:
@JacobTanenbaum I think we should just skip the complicated parsing here and just delete using the UUID.

Reply:
done